contracts/token-distribution/TokenDistribution.sol

```diff
@@ -19,17 +19,17 @@ abstract contract TokenDistribution {
 
     function getRoundTypeByKey(string memory _roundType) internal pure returns (RoundType) {
         bytes memory roundType = bytes(_roundType);
-        bytes32 Hash = keccak256(roundType);
+        bytes32 hash = keccak256(roundType);
 
-        if (Hash == keccak256("SEED") || Hash == keccak256("seed")) return RoundType.SEED;
-        if (Hash == keccak256("PRIVATE") || Hash == keccak256("private")) return RoundType.PRIVATE;
-        if (Hash == keccak256("PUBLIC") || Hash == keccak256("public")) return RoundType.PUBLIC;
-        if (Hash == keccak256("PLAYANDEARN") || Hash == keccak256("playandearn")) return RoundType.PLAYANDEARN;
-        if (Hash == keccak256("EXCHANGES") || Hash == keccak256("exchanges")) return RoundType.EXCHANGES;
-        if (Hash == keccak256("TREASURY") || Hash == keccak256("treasury")) return RoundType.TREASURY;
-        if (Hash == keccak256("ADVISOR") || Hash == keccak256("advisor")) return RoundType.ADVISOR;
-        if (Hash == keccak256("TEAM") || Hash == keccak256("team")) return RoundType.TEAM;
-        if (Hash == keccak256("SOCIAL") || Hash == keccak256("social")) return RoundType.SOCIAL;
+        if (hash == keccak256("SEED") || hash == keccak256("seed")) return RoundType.SEED;
```

Comment:
Why do these two calculations and comparisons?
The code should be more defensive - make it only check for lowercase instead, and be case sensitive? Because PlayAndEarn wouldn't work, for instance.

Reply:
Well in Solidity string comparison works differently as you cant just compare 2 strings with == or ===. You need to use keccak256 hash method to compare the hashes of the strings. Hence keccak256("TEAM") is not the same like keccak256("team"), thats why we have 2 comparisons.